array.ts

```diff
@@ -248,6 +248,48 @@ export const deleteAt = (i: number) =>
   <A>(as: ReadonlyArray<A>): O.Option<ReadonlyArray<A>> =>
     isOutOfBounds(i, as) ? O.none : O.some(unsafeDeleteAt(i, as));
 
+export const zipWith = <A, B, C>(
+  fa: ReadonlyArray<A>,
+  fb: ReadonlyArray<B>,
+  f: (a: A, b: B) => C,
+): ReadonlyArray<C> => {
+  const fc: Array<C> = [];
+  const len = Math.min(fa.length, fb.length);
+  for (let i = 0; i < len; i++) {
+    fc[i] = f(fa[i], fb[i]);
+  }
+  return fc;
+};
+
+export function zip<B>(
+  bs: ReadonlyArray<B>,
+): <A>(as: ReadonlyArray<A>) => ReadonlyArray<[A, B]>;
+export function zip<A, B>(
+  as: ReadonlyArray<A>,
+  bs: ReadonlyArray<B>,
+): ReadonlyArray<[A, B]>;
+export function zip<A, B>(
+  as: ReadonlyArray<A>,
+  bs?: ReadonlyArray<B>,
+): ReadonlyArray<[A, B]> | ((bs: ReadonlyArray<B>) => ReadonlyArray<[B, A]>) {
+  if (bs === undefined) {
+    return (bs) => zip(bs, as);
+  }
+  return zipWith(as, bs, (a, b) => [a, b]);
+}
```

Comment:
For now, I'd prefer we not overload any function implementations based around auto-curry. For this particular function I wouldn't curry it at all.

Reply:
Hello !
I removed overload and add tests for zip, but it still curried. (Tell me if you really want it without currying)
To do large array test I implemented range function.
I refactored zipWith like fp/ts 3.0.0